examples/xlnet/run.py

```diff
@@ -0,0 +1,249 @@
+# Copyright 2019 The Texar Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import argparse
+import importlib
+import logging
+import os
+import random
+import time
+from typing import Optional, Dict
+
+import numpy as np
+import sentencepiece as spm
+import texar as tx
+import torch
+import tqdm
+
+import xlnet
+
+
+def parse_args():
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "--config-data", required=True,
+        help="Path to the dataset configuration file.")
+
+    parser.add_argument(
+        "--mode", choices=["train", "eval"], default="train")
+    parser.add_argument(
+        "--checkpoint", type=str, default=None,
+        help="Path to a saved checkpoint file to load")
+    parser.add_argument(
+        "--save-dir", type=str, default="saved_models/",
+        help="Directory to save model checkpoints")
+
+    parser.add_argument(
+        "--pretrained", type=str,
+        default="pretrained/xlnet_cased_L-24_H-1024_A-16/xlnet_model.ckpt",
+        help="Path to the pretrained XLNet model checkpoint")
+    parser.add_argument(
+        "--data-dir", type=str, default=None,
+        help="Path to the directory containing raw data. "
+             "Defaults to 'data/<task name>'")
+    parser.add_argument(
+        "--cache-dir", type=str, default=None,
+        help="Path to the directory to cache processed data. "
+             "Defaults to 'processed_data/<task name>'")
+    parser.add_argument(
+        "--spm-model-path", type=str,
+        default="pretrained/xlnet_cased_L-24_H-1024_A-16/spiece.model",
+        help="Path to the sentencepiece model")
+    parser.add_argument(
+        "--uncased", type=bool, default=False,
+        help="Whether the pretrained model is an uncased model")
+
+    args = parser.parse_args()
+    config_module_path = args.config_data.replace('/', '.').replace('\\', '.')
+    if config_module_path.endswith(".py"):
+        config_module_path = config_module_path[:-3]
+    config_data = importlib.import_module(config_module_path)
```

Comment:
Line.68--71 can be wrapped as a util function. Quite a few examples need this (including the ones in texar-TF)

Reply:
Done. However, this was intended as a temporary workaround. In future, I think we can have per-example `default_hparams`, and use the `HParams` interface for sanity checks & default value filling.